### PR TITLE
TE-243 HTML Widget

### PR DESCRIPTION
**Pull request by davidpelayo**
[Related YouTrack issue](https://youtrack.lodgify.net/issue/TE-243)

### What **one** thing does this PR do?
- Implements the HTML widget

### Other Notes
- It sanitizes the HTML received not allowing to inject scripts, style tags, iframes, among others.

### CODE REVIEW

File: src/components/widgets/HTML/Readme.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+<HTML htmlString={headings} />
+```
+
+### Variations
+
+#### Nesting
+
+```jsx
+<HTML htmlString="<h2>Inception Level 0 😱</h2><br />">
+  <HTML htmlString="<h3>Incetion Level 1 😱</h3><br />">
```
Comment (marcdavi-es) [inline src/components/widgets/HTML/Readme.md]: `Inception` -> `Inception`

File: src/components/widgets/HTML/Readme.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    <HTML htmlString="<h4>Incetion Level 2 😱</h4><br />">
```
Comment (marcdavi-es) [inline src/components/widgets/HTML/Readme.md]: and here

File: src/components/widgets/HTML/component.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import DOMPurify from 'dompurify';
+
+/**
+ * The standard HTML Widget
```
Comment (marcdavi-es) [inline src/components/widgets/HTML/component.js]: Something more descriptive?

`The HTML widget sanitises and renders HTML strings.`  or whatever?

File: src/components/widgets/HTML/component.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ * @returns {Object}
+ */
+export const Component = ({ htmlString, children }) =>
+  children ? (
```
Comment (marcdavi-es) [inline src/components/widgets/HTML/component.js]: Hmm 🤔 ... as a consumer I think I would expect 

```
<HTML htmlString="<picture></picture>">
   <img src="fridge.jpeg" />
</HTML>
```

to render 

```
<picture>
   <img src="fridge.jpeg" />
</picture>
```
what do you think?

File: src/components/widgets/HTML/component.spec.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { mount, shallow } from 'enzyme';
+import DOMPurify from 'dompurify';
+
+import { Component as HTMLWidget } from './component';
+
+const { headings } = require('./mock-data/examples');
+
+describe('<HTML />', () => {
+  it('should render a single Lodgify UI `HTML` component', () => {
```
Comment (marcdavi-es) [inline src/components/widgets/HTML/component.spec.js]: isn't this testing whether or not `enzyme.shallow` works?

File: src/components/widgets/HTML/component.js (hunk 4 above)
Comment (davidpelayo) [reply src/components/widgets/HTML/component.js]: Really? I can't believe the way you think dude haha!
We see reality in total different ways. I can't imagine any example whereas given a encoded (or decoded) HTML string as input, renders the output as you described 🤣

Comment (davidpelayo) [reply src/components/widgets/HTML/component.js]: Maybe is just I over complicated it. Actually there is no need to render children. Only a given HTML string.

File: src/components/widgets/HTML/component.spec.js (hunk 5 above)
Comment (davidpelayo) [reply src/components/widgets/HTML/component.spec.js]: I'm sorry not to understand what you mean?

File: src/components/widgets/HTML/component.js (hunk 3 above)
Comment (davidpelayo) [reply src/components/widgets/HTML/component.js]: Agreed.

File: src/components/widgets/HTML/component.spec.js (hunk 5 above)
Comment (davidpelayo) [reply src/components/widgets/HTML/component.spec.js]: Ok, I see now...

**Pull request merged**